src/rand.rs

```diff
@@ -140,6 +140,28 @@ mod sysrand {
 }
 
 // Keep the `cfg` conditions in sync with the conditions in lib.rs.
+#[cfg(target_os = "redox")]
```

Comment:
Rather than duplicate the code for the `urandom` submodule, let's abstract out the one difference, the path to the random file, and share the rest of the code.